src/backend/x11/mod.rs

```diff
@@ -0,0 +1,563 @@
+                                        // to match libinput.
+                                        key: key_press.detail as u32 - 8,
+                                        count: key_counter.fetch_add(1, Ordering::SeqCst) + 1,
+                                        state: KeyState::Pressed,
+                                    },
+                                }),
+                                &mut event_window,
+                            )
+                        }
+                    }
+
+                    x11::Event::KeyRelease(key_release) => {
+                        if key_release.event == window.inner {
+                            // atomic u32 has no checked_sub, so load and store to do the same.
+                            let mut key_counter_val = key_counter.load(Ordering::SeqCst);
+                            key_counter_val = key_counter_val.saturating_sub(1);
+                            key_counter.store(key_counter_val, Ordering::SeqCst);
+
+                            callback(
+                                Input(InputEvent::Keyboard {
+                                    event: X11KeyboardInputEvent {
+                                        time: key_release.time,
+                                        // It seems as if X11's keycodes are +8 relative to the libinput
+                                        // keycodes that are expected, so subtract 8 from each keycode
+                                        // to match libinput.
+                                        key: key_release.detail as u32 - 8,
+                                        count: key_counter_val,
+                                        state: KeyState::Released,
+                                    },
+                                }),
+                                &mut event_window,
+                            );
+                        }
+                    }
+
+                    x11::Event::MotionNotify(motion_notify) => {
+                        if motion_notify.event == window.inner {
+                            // Use event_x/y since those are relative the the window receiving events.
+                            let x = motion_notify.event_x as f64;
+                            let y = motion_notify.event_y as f64;
+
+                            callback(
+                                Input(InputEvent::PointerMotionAbsolute {
+                                    event: X11MouseMovedEvent {
+                                        time: motion_notify.time,
+                                        x,
+                                        y,
+                                        size: window.size(),
+                                    },
+                                }),
+                                &mut event_window,
+                            )
+                        }
+                    }
+
+                    x11::Event::ResizeRequest(resized) => {
```

Comment:
Uhm. Does this code ever trigger? You are not a window manager, you should not get ResizeRequest events. Instead, you should get ConfigureNotify events.

Reply:
Seems to fire under KDE, not sure if KDE has bungled their WM

Reply:
https://www.x.org/releases/X11R7.6/doc/xproto/x11protocol.html#events:ResizeRequest 

> This event is reported to the client selecting ResizeRedirect on the window [...]

Oh, hey, you *are* a window manager. You are including `RESIZE_REDIRECT` in your event mask in WIndowInner. I missed that earlier. You should instead use `ConfigureNotify` events which are sent from `STRUCTURE_NOTIFY` which you already include.

What `ResizeRedirect` does: It basically makes your window "un-resizable". Only you can resize your own window.